src/frontend/planner_test/tests/testdata/output/append_only.yaml

```diff
@@ -25,9 +25,10 @@
     select v1 from t1 order by v1 limit 3 offset 3;
   stream_plan: |-
     StreamMaterialize { columns: [v1, t1._row_id(hidden)], stream_key: [t1._row_id], pk_columns: [v1, t1._row_id], pk_conflict: NoCheck }
-    └─StreamTopN [append_only] { order: [t1.v1 ASC], limit: 3, offset: 3 }
-      └─StreamExchange { dist: Single }
-        └─StreamTableScan { table: t1, columns: [t1.v1, t1._row_id], stream_scan_type: ArrangementBackfill, stream_key: [t1._row_id], pk: [_row_id], dist: UpstreamHashShard(t1._row_id) }
+    └─StreamExchange { dist: HashShard(t1.v1) }
+      └─StreamTopN [append_only] { order: [t1.v1 ASC], limit: 3, offset: 3 }
+        └─StreamExchange { dist: Single }
+          └─StreamTableScan { table: t1, columns: [t1.v1, t1._row_id], stream_scan_type: ArrangementBackfill, stream_key: [t1._row_id], pk: [_row_id], dist: UpstreamHashShard(t1._row_id) }
```

Comment:
I do not know if we should make some special behavior for the `order by` clause with `limit`... Maybe it is ok because there are not many people write a ToN streaming job and has expectation on its performance

Reply:
+1